functions/__tests__/util.test.ts

```diff
@@ -0,0 +1,217 @@
+  const firebaseMock = firebaseFunctionsTestInit()
+
+  test('return id after create document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      {},
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id).toEqual('2')
+  })
+
+  test('return id after update document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bars' },
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id).toEqual('2')
+  })
+
+  test('return id after delete document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      {},
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id).toEqual('1')
+  })
+})
+
+describe('getSearchableFields', () => {
+  let util
+  let config
+  let restoreEnv
+  let mockGetSearchableFields
+
+  beforeEach(() => {
+    jest.resetModules()
+    restoreEnv = mockedEnv(environment)
+    config = global.config
+  })
+  afterEach(() => restoreEnv())
+
+  test('configuration detected from environment variables', () => {
+    const mockConfig = config()
+    expect(mockConfig).toMatchSnapshot({})
+  })
+
+  test('return empty list', () => {
+    util = require('../src/util')
+    mockGetSearchableFields = util.getSearchableFields()
+    expect(mockGetSearchableFields).toMatchObject([])
+  })
+
+  test('return list with one field', () => {
+    restoreEnv = mockedEnv({ ...environment, SEARCHABLE_FIELDS: 'field' })
+    util = require('../src/util')
+    mockGetSearchableFields = util.getSearchableFields()
+    expect(mockGetSearchableFields).toMatchObject(['field'])
+  })
+
+  test('return list with multiple field', () => {
+    restoreEnv = mockedEnv({
+      ...environment,
+      SEARCHABLE_FIELDS: 'field1,field2,field3',
+    })
+    util = require('../src/util')
+    mockGetSearchableFields = util.getSearchableFields()
+    expect(mockGetSearchableFields).toMatchObject([
+      'field1',
+      'field2',
+      'field3',
+    ])
+  })
+
+  test('return list with multiple field and space', () => {
+    restoreEnv = mockedEnv({
+      ...environment,
+      SEARCHABLE_FIELDS: 'field1, field2,  field3',
+    })
+    util = require('../src/util')
```

Comment:
Could the require be the first line of the function?

Reply:
No, it didn't work, it should be load after the environment. But I think it is not a good way to make this kind of test but I didn't find a better way for now.

Reply:
Could you make an issue about it?